### ISSUE TITLE
Vendor assets are placed at the bottom of the output bundle

### ISSUE THREAD

**Issue by friday** [brunch/brunch#1734]
### Description

As I understand it, `conventions.vendor` is used to determine when modules should be wrapped, and pushing them to the top of the output bundle.

The logic of the latter is seemingly inversed for in this case, putting them at the bottom instead.

### brunch config

```javascript
module.exports = {
  // Make bundles easier to read
  modules: {definition: false},
  npm: {enabled: false},

  paths: {watched: ['client']},
  conventions: {
    // This is just to clarify, the default vendor conventions actually
    // does the exact same for this example
    vendor: path => path.includes('vendor')
  },
  files: {
    javascripts: {
      order: {
        before: path => path.includes('start'),
        after: path => path.endsWith('jquery.js')
      },
      joinTo: {
        'js/hello.min.js': [
          'client/js/vendor/jquery.js',
          'client/js/app/hello.js',
          'client/js/vendor/lodash.js',
          'client/js/app/start.js'
        ]
      }
    }
  }
};
```

### Behavior
Conventions.vendor works as intended for wrapping, but puts vendors assets at bottom instead of top (my assets are just logging their file name for testing purposes):
```
require.register("client/js/app/start.js", function(exports, require, module) {
console.log('start.js')

});

;require.register("client/js/app/hello.js", function(exports, require, module) {
console.log('hello.js')

});

;console.log('lodash.js')

;console.log('jquery.js')

;
//# sourceMappingURL=hello.min.js.map
```

### Expected behavior

* Vendor files should be pushed to the top, not bottom
* `order.before` should take precedence over vendor assets. According to the documentation the ordering should be `[before] -> [bower] -> [vendor] -> [everything else] -> [after]`

### Environment

1. Brunch: 2.10.10
1. Node.js: 8.3.0
1. Yarn: 0.27.5
1. Operating system: Darwin (os x)

### Workaround

Set `conventions.vendor` to `() => false` or `() => true` (if you want to wrap all or no asset)

### Replication
Repo: https://github.com/friday/brunch-1734

**Comment by cxw42** [comment]
I believe this is true.  The lines [here](https://github.com/brunch/brunch/blob/f437170721a2acd0dc7d71a98ecb69fedcf2bc6b/lib/fs_utils/generate.js#L112) seem to be flipped - if I understand the docs correctly, `nonModuleFiles` should be output before `moduleFiles`.

**Comment by cxw42** [comment]
Looks like this is a duplicate of #1527.

**Comment by friday** [comment]
Thanks! Indeed looks like the same issue. Closing this one then :)
